dragging.js

```diff
@@ -0,0 +1,86 @@
+const leftSide = resizer.previousElementSibling;
+const rightSide = resizer.nextElementSibling;
+
+// The current position of mouse
+let x = 0;
+let y = 0;
+
+// Width of left side
+let leftWidth = 0;
+leftSide.style.width = optimizeWidth(); //Set a good looking default width each time the page loads.
+
+// Handle the mousedown event
+// that's triggered when user drags the resizer
+const mouseDownHandler = function (e) {
+    // Get the current mouse position
+    x = e.clientX;
+    y = e.clientY;
+    leftWidth = leftSide.getBoundingClientRect().width;
+
+    // Attach the listeners to `document`
+    document.addEventListener('mousemove', mouseMoveHandler);
+    document.addEventListener('mouseup', mouseUpHandler);
+};
+
+// Attach the handler
+resizer.addEventListener('mousedown', mouseDownHandler);
+
+const mouseMoveHandler = function (e) {
+    // How far the mouse has been moved
+    const dx = e.clientX - x;
+    const dy = e.clientY - y;
+
+    const newLeftWidth = ((leftWidth + dx) * 100) / resizer.parentNode.getBoundingClientRect().width;
+    leftSide.style.width = `${newLeftWidth}%`;
+
+	resizer.style.cursor = 'col-resize';
+	document.body.style.cursor = 'col-resize';
+
+	leftSide.style.userSelect = 'none';
+    leftSide.style.pointerEvents = 'none';
+
+    rightSide.style.userSelect = 'none';
+    rightSide.style.pointerEvents = 'none';
+};
+
+const mouseUpHandler = function () {
+    resizer.style.removeProperty('cursor');
+    document.body.style.removeProperty('cursor');
+
+    leftSide.style.removeProperty('user-select');
+    leftSide.style.removeProperty('pointer-events');
+
+    rightSide.style.removeProperty('user-select');
+    rightSide.style.removeProperty('pointer-events');
+
+    // Remove the handlers of `mousemove` and `mouseup`
+    document.removeEventListener('mousemove', mouseMoveHandler);
+    document.removeEventListener('mouseup', mouseUpHandler);
+};
+
+//takes the user's screen dimensions and type of page (guide or nirnroute currently) and generates the best starting width.
+//NEEDS TO RETURN A STRING.
+function optimizeWidth(){
+    let page = location.pathname.split("/").slice(-1);
+    let width = "";
+    //We can add a check in here to skip the automatic optimizations if iframeWidth setting has a value now.
+
+    //Automatic widths if no settings for iframeWidth exist.
+    if(page == "nirnroute.html"){
+        width = "67.5%"; //Currently only good for a 1920x1080 aspect ratio, will make better later.
+    }
+    else if(page == "speedrun-4.html" || page == "speedrun.html" || page == "casual.html"){ //TODO: Get rid of speedrun-4 check when merged into the version with redirect pages.
+        width = "55%"; //Currently only good for a 1920x1080 aspect ratio, will make better later.
+    }
+    else {
+        width = "55%"; //a generally decent looking default. Eventually want to programmatically determine if a page is a "guide" page or "map" page.
+    }
+    return width;
```

Comment:
I think this is something better done in the html pages themselves. Just set the default width in the page itself instead of creaing a method here.

Reply:
My long term plan with this function is that it takes in other parameters and provides a more nuanced starting width based on things like the window aspect ratio and whatnot. For example, having the nirnroute always start with the pictures framed as large as possible unless it were to take up over 50% of the map and things like that. imo this is a much more future proofed option than setting a default width in the html page.

Reply:
no plz no
it is so much harder to find the "right" width for every possible combination of zoom + pixel density + screen width + screen height + default font size + browser
The best option is to keep it simple, provide a "good enough" default for like, 1080p fullscreen and default mobile, and then leave the rest up to the user.

Reply:
I've changed it, although tbh I'm not super thrilled with the workflow being that we have to go into each individual html page and find/change this style value. I might make a class for each starting width in the future.